qcodes_contrib_drivers/drivers/Attocube/ANC350.py

```diff
@@ -0,0 +1,510 @@
+
+    def disable_auto_move(self) -> None:
+        """Disables automatic moving"""
+        self._parent._lib.start_auto_move(self._parent._device_handle, self._axis, False, False)
+
+    @classmethod
+    def _map_relative_parameter(cls, relative: Optional[Union[bool, str]]) -> bool:
+        if relative is None:
+            return False
+        if relative in [False, True]:
+            return bool(relative)
+
+        if relative in cls._relative_mapping:
+            return cls._relative_mapping[relative]
+
+        raise ValueError("Unexpected value for argument `relative`. Allowed values are: {}".format(
+            [None, False, True, *cls._relative_mapping.keys()]))
+
+    def _get_position(self) -> float:
+        """
+        Get the current position of this axis
+
+        Returns:
+            Current position in millimeters [mm] (linear type actuators) or millidegrees [m°]
+            (goniometers and rotators)
+        """
+        # Conversion from meters (degrees) to millimeters (millidegrees) because the wrapper works
+        # with meters (degrees)
+        return self._parent._lib.get_position(self._parent._device_handle, self._axis) * 1e3
+
+    def _set_position(self, position: float) -> None:
+        """(EXPERIMENTAL FUNCTION)
+        The axis moves to the given position with the target range that is set before.
+
+        Args:
+            position: The position the axis moves to
+        """
+        self._set_target_position(position)
+        self._set_output(2)  # 2 = "auto"
+
+    def _get_frequency(self) -> float:
+        """
+        Returns the frequency parameter of this axis.
+
+        Returns:
+            Frequency in Hertz [Hz], internal resolution is 1 Hz
+        """
+        return self._parent._lib.get_frequency(self._parent._device_handle, self._axis)
+
+    def _set_frequency(self, frequency: float) -> None:
+        """
+        Sets the frequency parameter for this axis
+
+        Args:
+            frequency (float): Frequency in Hertz [Hz], internal resolution is 1 Hz
+        """
+        frequency = int(round(frequency))
+
+        self._parent._lib.set_frequency(self._parent._device_handle, self._axis, frequency)
+
+    def _get_amplitude(self) -> float:
+        """
+        Returns the amplitude parameter of this axis.
+
+        Returns:
+            Amplitude in Volts [V]
+        """
+        return self._parent._lib.get_amplitude(self._parent._device_handle, self._axis)
+
+    def _set_amplitude(self, amplitude: float) -> None:
+        """
+        Sets the amplitude parameter for an axis
+
+        Args:
+            amplitude: Amplitude in Volts [V] (internal resolution is 1mV)
+        """
+        self._parent._lib.set_amplitude(self._parent._device_handle, self._axis, amplitude)
+
+    def _get_status(self) -> Dict[str, bool]:
+        """
+        Reads status information about an axis
```

Comment:
How does the user know which axis this status belong to? Is it returning the current axis? Then shouldn't we return that axis number?

Reply:
Because of object orientation. This function belongs to class `Anc350Axis`, which represents a single axis. Which axis is represented depends on which axis number (and name) is passed in the constructor. So the user can check e.g. `Anc350Axis.name` or `short_name` if he wants to know which axis he is working with. Additionally, I'm pretty sure that most users only use the top-level instrument `Anc350`. So, the user would call e.g. `anc350.x_axis.status()`. Thus it's pretty clear that the user is requesting the status of `x_axis`.

Reply:
Thanks